### PR TITLE
add docker-env and podman-env to minikube status

### LINKED ISSUE #6535: feature: minikube status- indicate if using "docker-env"
one of the common things I do after eval $(minikube docker-env) is checking if I am using my own docker or minikube's docker

it would be nice if "minikube status" command would tell me 

```
host: ...
kubelet: ..
apiserver: ...
kubeconfig: ...
docker-env: in-use
```
and if we are not using docker-env no need to add that line and keep it hidden.
but we could includ that in the json output.

### PR DESCRIPTION
**Pull request by tharun208**
Signed-off-by: Tharun <rajendrantharun@live.com>

Before: `minikube status`
Screen Shot 2021-03-23 at 2 36 18 PM

After: `minikube docker-env` and `minikube-status` and `minikube docker-env --unset`
Screen Shot 2021-03-23 at 2 37 17 PM

`minikube status --output=json` result:
Screen Shot 2021-03-23 at 3 36 54 PM

Fixes #6535

### CODE REVIEW

**Review by medyagh — CHANGES_REQUESTED** [review]
lets add same thing for podman-env

**Comment by medyagh** [comment]
Do you mind pasting the out of minikube status --output=json
And can we ensure the json output has it too?

File: pkg/minikube/config/types.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ type ClusterConfig struct {
 	ListenAddress           string   // Only used by the docker and podman driver
 	Network                 string   // only used by docker driver
 	MultiNodeRequested      bool
+	DockerEnvInUse          bool
```
Comment (medyagh) [inline pkg/minikube/config/types.go:84]: This should not be part of config. Because in one terminal it might be in use and other terminal not but config is global. So let's remove it

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/config/types.go (hunk 1 above)
Comment (tharun208) [reply pkg/minikube/config/types.go:84]: @medyagh, without the config change. can you help me with how we can implement this for podman?

for docker, we can make use of this `MINIKUBE_ACTIVE_DOCKERD` env variable at runtime to check if it is the same as the cluster name.

Comment (medyagh) [reply pkg/minikube/config/types.go:84]: For podman u could see the podman-env.go file

But also lets make sure the minikbue Config does not have that DockerEnvInUse ...

**Comment by tharun208** [comment]
@medyagh, I updated the code without making any changes in local. But, I am not able to run the `functional test` suite successfully locally.

File: cmd/minikube/cmd/docker-env.go [hunk 2]
```diff
@@ -252,7 +252,8 @@ var dockerEnvCmd = &cobra.Command{
 		sh := shell.EnvConfig{
 			Shell: shl,
 		}
```
Comment (medyagh) [inline cmd/minikube/cmd/docker-env.go:254]: Any reason this was moved up ? If not revert the unrelated changes

File: cmd/minikube/cmd/status.go [hunk 3]
```diff
@@ -136,6 +137,8 @@ type Status struct {
 	Kubeconfig string
 	Worker     bool
 	TimeToStop string
```
Comment (medyagh) [inline cmd/minikube/cmd/status.go:139]: Would u plz paste the output of minikube status —outpus=json after this PR ?
Does this field reflect in the json output ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/status.go (hunk 3 above)
Comment (tharun208) [reply cmd/minikube/cmd/status.go:139]: @medyagh, I added the output for `minikube status —outpus=json` in the pr description

**Review by medyagh — APPROVED** [review]

**Comment by medyagh** [comment]
thank you for this contribution @tharun208
